### PR TITLE
Improve Lograge configs

### PR DESCRIPTION
**Pull request by santib**
TODO:
- [x] Compare it with [semantic_logger](https://logger.rocketjob.io/rails.html)
- [x] Check if it's ok to log IPs (GDPR)
- [x] Check if there is something else to log
- [x] Check log levels
- [x] Check Rails defaults

Notes:
- [semantic_logger](https://logger.rocketjob.io/rails.html) is definitively more powerful. But I'm not sure it's worth the effort, just the setup requires considering many things like Heroku, Puma, RSpec, etc

Sources:
- https://prefab.cloud/blog/rails-logging-gems-2024/
- https://betterstack.com/community/guides/logging/best-ruby-logging-libraries/

### CODE REVIEW

File: config/initializers/filter_parameter_logging.rb [hunk 1]
```diff
@@ -6,5 +6,5 @@
 # Use this to limit dissemination of sensitive information.
 # See the ActiveSupport::ParameterFilter documentation for supported notations and behaviors.
 Rails.application.config.filter_parameters += %i[
-  passw secret token _key crypt salt certificate otp ssn
+  passw email secret token _key crypt salt certificate otp ssn cvv cvc
```
Comment (santib) [inline config/initializers/filter_parameter_logging.rb:9]: taken from https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/config/initializers/filter_parameter_logging.rb.tt#L7

File: config/initializers/lograge.rb [hunk 2]
```diff
@@ -2,14 +2,17 @@
 
 Rails.application.configure do
   config.lograge.enabled = true
+  config.lograge.base_controller_class = ['ActionController::API', 'ActionController::Base']
```
Comment (santib) [inline config/initializers/lograge.rb:5]: API request weren't being handled by `lograge`

File: config/initializers/lograge.rb [hunk 3]
```diff
@@ -2,14 +2,17 @@
...
   config.lograge.custom_payload do |controller|
     request = controller.request
     {
+      current_user_id: controller.current_user&.id,
```
Comment (santib) [inline config/initializers/lograge.rb:9]: log `current_user` if logged in. it doesn't work nicely for `current_admin_user` when accessing API endpoints (can investigate further if necessary)

File: config/initializers/lograge.rb [hunk 4]
```diff
@@ -2,14 +2,17 @@
...
       host: request.host,
       path: request.path,
       query_string: request.query_string,
       referer: request.referer,
-      user_agent: request.user_agent
+      user_agent: request.user_agent,
+      remote_ip: request.remote_ip
```
Comment (santib) [inline config/initializers/lograge.rb:15]: I think logging the IP is fine since it doesn't identify a person

**Review by JulianPasquale — APPROVED** [review]

**Review by PerezIgnacio — APPROVED** [review]

**Pull request merged**